providers/keap/parse.go

```diff
@@ -15,3 +17,70 @@ func makeNextRecordsURL(moduleID common.ModuleID) common.NextPageFunc {
 		return jsonquery.New(node).StrWithDefault("next_page_token", "")
 	}
 }
+
+// Before parsing the records if applicable API will be called one time to get model describing custom fields.
+// Object will then be enhanced using model.
+func (c *Connector) parseReadRecords(
+	ctx context.Context, config common.ReadParams, jsonPath string,
+) common.RecordsFunc {
+	return func(node *ajson.Node) ([]map[string]any, error) {
+		arr, err := jsonquery.New(node).Array(jsonPath, true)
+		if err != nil {
+			return nil, err
+		}
+
+		customFields, err := c.requestCustomFields(ctx, config.ObjectName)
+		if err != nil {
+			return nil, err
+		}
+
+		result := make([]map[string]any, len(arr))
+
+		for index, object := range arr {
+			item, err := c.enhanceObjectWithCustomFieldNames(object, customFields)
+			if err != nil {
+				return nil, err
+			}
+
+			result[index] = item
+		}
+
+		return result, nil
+	}
+}
+
+// In general this does the usual JSON parsing.
+// However, those objects that contain "custom_fields" are processed as follows:
+// * Locate custom fields in JSON read response.
+// * Replace ids with human-readable names, which is provided as argument.
+// * Place fields at the top level of the object.
+func (c *Connector) enhanceObjectWithCustomFieldNames(
+	node *ajson.Node,
+	fields map[int]modelCustomField,
+) (map[string]any, error) {
+	object, err := jsonquery.Convertor.ObjectToMap(node)
+	if err != nil {
+		return nil, err
+	}
+
+	if len(fields) == 0 {
+		// no custom fields, the object is ready as is.
+		return object, nil
+	}
+
+	customFieldsResponse, err := jsonquery.ParseNode[readCustomFieldsResponse](node)
+	if err != nil {
+		return nil, err
+	}
+
+	// Replace identifiers with human-readable field names which were found by making a call to "/model".
+	for _, field := range customFieldsResponse.CustomFields {
+		if model, ok := fields[field.ID]; ok {
+			object[model.FieldName] = field.Content
+		}
+	}
+
+	delete(object, "custom_fields")
```

Comment:
Can you keep `custom_fields`? If a builder is going by the API docs, I wouldn't want them to be confused about why there's missing information in the raw object. As a rule of thumb, please add information, not remove / replace

Reply:
Sure. Other connectors have flattening of nested objects, and in those cases I had removed the nested object. Recent example would be Klaviyo:
```
	    {
	        "type": "",
	        "id": "",
	        "attributes": {
	            "test_account": false,
	            "contact_information": {},
	            "locale": ""
	        },
	        "links": {}
	    }
	],
```
Attributes object removed and it's properties moved to parent.
Should those cases be addressed as conceptually it is similar?

Reply:
Since this object is what we also send to the user as the 'Raw' object later on, I'd want to have at least a copy of the original untouched response. For Klaviyo, IMO we should still have the attributes object to maintain the original payload signature, but I'll let @laurenzlong take the call since this is builder facing